bubble-wrap-style.yaml

```diff
@@ -2215,15 +2258,40 @@ layers:
             draw:
                 mapzen_icon_library:
                     # you need to match any custom shield to the vector tile `network` values
+                    # sprite: |
+                    #     function() {
+                    #       if( feature.shield_text ) {
+                    #         return ['generic_shield-', feature.shield_text.length, 'char'].join('');
+                    #       }
+                    #       return ['generic_shield-', feature.ref.length, 'char'].join('');
+                    #     }
+                    ## [].join('') is good - Better build no new strings at all:
+                    sprite_default: ~ # remove 'generic' default to make discard label-by-sprite function work
                     sprite: |
                         function() {
-                            if( feature.shield_text ) {
-                                return ('generic_shield-' + feature.shield_text.length + 'char');
-                            } else {
-                                return ('generic_shield-' + feature.ref.length + 'char');
+                          var text = feature.shield_text;
+                          if (text === undefined) {
+                            text = feature.ref;
+                            if (text === undefined) { return undefined; }
+                            if (text.length > 3) {
```

Comment:
This should only be done if the text is coming from `ref` not `shield_text`, as `shield_text` deliberately sometimes keeps whitespace in the value.

But later on since you propose to set the text to just ref, the symbol size will no longer match. So we either need to stop stripping the space from `ref` **here**, or add the stripping to the text source below. I'd be fine stopping stripping of the space from ref here as the vector data is soooo much better starting in Tilezen v1.5 tiles.